### DISCUSSION TITLE
How to handle onClick outside of the layer in v7

### DISCUSSION THREAD

**Discussion by f1sh1918** [uber/react-map-gl#1809] [Q&A]
Hi all,

i recently upgraded from v6 to v7 and i recognized that the onClick event changed.
In v6 i checked `onClick` if a feature of my layer (i'm using interactiveLayerIds) was clicked or not and select/deselect the feature.
Like in google maps if you click somewhere on the map the selected feature gets deselected.
```js
onClick: (event: [MapLayerMouseEvent](https://visgl.github.io/react-map-gl/docs/api-reference/types#maplayermouseevent)) => void
Called when a pointing device (usually a mouse) is pressed and released at the same point on the map.

If interactiveLayerIds is specified, the event will fire only when the point that is clicked twice contains a visible portion of the specifed layer.
```
Now i'm facing the problem that i can not deselect the feature because the onClick event is only fired if click on a canvas of my layer. 
Unfortunately i didn't find a callback function that recognizes or catches when i don't click on a feature of my layer.
Have you any hint for me how to implement that functionality with react-maps-gl v7

**Comment by f1sh1918** [comment]
for now i can work with wrapper and set there onClick for deselection and stopPropagation on the onClick Event of the map. But this is not very nice to be honest
` event.originalEvent.stopPropagation()`

**Comment by fpsampayo** [comment]
Currently, I do this disabling `interactiveLayersIds` and using the maplibre/mapbox's `queryRenderedFeatures` with the `point` that the [`MapLayerMouseEvent`](https://visgl.github.io/react-map-gl/docs/api-reference/types#maplayermouseevent) returns.
